src/io/flutter/run/common/CommonTestConfigUtils.java

```diff
@@ -26,45 +32,76 @@
  */
 public abstract class CommonTestConfigUtils {
   /**
-   * Regex that matches customized versions of the Widget test function from package:flutter_test/src/widget_tester.dart.
-   * <p>
-   * This will match all test methods with names that start with 'test', optionally
-   * have additional text in the middle, and end with 'Widgets'.
+   * How the Dart Analysis Server identifies runnable tests in the {@link Outline} it generates.
    */
-  public static final Pattern WIDGET_TEST_REGEX = Pattern.compile("test([A-Z][A-Za-z0-9_$]*)?Widgets");
+  private static final String UNIT_TEST_TEST = "UNIT_TEST_TEST";
 
-  public abstract TestType asTestCall(@NotNull PsiElement element);
+  /**
+   * How the Dart Analysis Server identifies runnable test groups in the {@link Outline} it generates.
+   */
+  private static final String UNIT_TEST_GROUP = "UNIT_TEST_GROUP";
 
   public static String convertHttpServiceProtocolToWs(String url) {
     return StringUtil.trimTrailing(
       url.replaceFirst("http:", "ws:"), '/') + "/ws";
   }
 
+  public abstract TestType asTestCall(@NotNull PsiElement element);
+
+  /**
+   * Gets the elements from the outline that are runnable tests.
+   */
+  protected Map<DartCallExpression, TestType> getTestsFromOutline(@NotNull PsiFile file) {
```

Comment:
We're creating a map and only using it to look up a single item as far as I can tell. Either simplify to just lookup the one item you care about directly or cache the map invalidating when outlines change.

Reply:
The map call is used in findNamedTestCall below.  IntelliJ might give us an arbitrary PsiElement and ask us to find the test call that is parent to it.

This means we need to look up the tree to find an element that is in the collection of all test calls.